src/graphnet/data/extractors/icecube/i3calorimetry.py

```diff
@@ -0,0 +1,237 @@
+"""Extract all the visible particles entering the volume."""
+
+from typing import Dict, Any, TYPE_CHECKING, Tuple
+
+from .utilities.gcd_hull import GCD_hull
+from .i3extractor import I3Extractor
+
+import numpy as np
+
+from graphnet.utilities.imports import has_icecube_package
+
+if has_icecube_package() or TYPE_CHECKING:
+    from icecube import (
+        icetray,
+        dataclasses,
+        MuonGun,
+        simclasses,
+    )  # pyright: reportMissingImports=false
+
+
+class I3Calorimetry(I3Extractor):
+    """Energy on appearance of all visible particles in the volume."""
```

Comment:
Since this extracts the energy on appearance, maybe a more suitable name for the extractor would be `I3EnergyOnAppearanceExtractor`, or something similar?

Reply:
The discerning difference between the `I3Calorimetry` approach and the `I3HighestEparticleExtractor` is that the `I3Calorimetry` approach tries to look at everything that enters the detector volume (only daughters of the primary if daughters=True). Whereas, the `I3HighestEparticleExtractor` only tries to reconstruct the _single_ highest energy detectable particle entering the detector. The _calorimetric_ approach provides several different types of energy labels which can be reconstructed and therefor energy on appearance. I will try to update the docstring to make this more clear.